docs/proposals/05-object-storage.md

```diff
@@ -0,0 +1,146 @@
+---
+title: Object Storage
+
+oep-number: 5
+
+creation-date: 2022-12-19
+
+status: implementable
+
+authors:
+
+- @lukasfrank
+- @gehoern
+
+reviewers:
+
+- @adracus
+- @MalteJ
+
+---
+
+# OEP-5: Object Storage
+
+## Table of Contents
+
+- [Summary](#summary)
+- [Motivation](#motivation)
+    - [Goals](#goals)
+    - [Non-Goals](#non-goals)
+- [Proposal](#proposal)
+- [Alternatives](#alternatives)
+
+## Summary
+Object storage builds the basis for many cloud applications. An Object Storage provides a simplified object 
+model for files, but has a reduced set of security and access features (non-posix). This functionality is built on top 
+of the HTTP protocol. The current market standard is [S3](https://docs.aws.amazon.com/AmazonS3/latest/API/Type_API_Reference.html). 
+This document describes how to integrate simplified S3 buckets into the OnMetal API without taking too many details of 
+the S3 feature completeness itself.
+
+## Motivation
+Object Storage is demanded by cloud native applications, therefore, OnMetal needs to provide it for a complete solution.
+The Object Storage service should be integrated into OnMetal and is not designed 
+to be _just_ a service on top of OnMetal. The used protocol is called [S3](https://docs.aws.amazon.com/AmazonS3/latest/API/Type_API_Reference.html)
+which introduces a storage entity called bucket. For the beginning only the bucket creation and removal is covered.
+
+### Goals
+- S3-compatible object storage implementation
+- Automatic assigned public storage endpoint 
+- Providing a REST-API endpoint to address the bucket
+
+### Non-Goals
+- Support other object storage protocols than S3
+- Internal buckets (reachable only from inside the cluster)
+- Quota handling (size or object number limitation)
+- External Object level access control (beyond what the S3 implementation provides)
+
+## Proposal
+The proposal to provide an Object Storage consists of three API resources: `Bucket`, `BucketClass` and `BucketPool`. 
+A `Bucket` is the S3 enabled storage endpoint. IOPS/Bandwidth limitations are controlled via a `BucketClass` and the 
+capabilities of the underlying storage provider are expressed via a `BucketPool`. A `Bucket` can be requested 
+from a `BucketPool` as long as it can provide the performance characteristics described in the `BucketClass`. 
+The proposed API resources are similar to `Volume`, `VolumeClass` and `Volumepool` except that a volume is a 
+block device with a specific driver.
+
+
+### Bucket
+
+A `Bucket` is a *namespaced* resource to request S3-compatible object storage. 
+`Bucket`s with the `type` `Public` set (only valid type for now) are accessible from the internet.
```

Comment:
imho the `public` setting is an ACL setting and should probably not be configured on the bucket itself. I'd expect to set this permission via the S3 API! Please check.

Reply:
The `type` `Public` describes if the bucket should be reachable from the internet and gives us a hint what we need to provision in the background. The `ACL` settings are done via the S3 API.

Reply:
I don't understand what it means "reachable from the internet". Why do we need to put this into the bucket spec?

Reply:
How is it different from the ACLs?

Reply:
That is handled by the S3 protocol afaik. I'd like to remove this parameter if we don't know what it's for.

Reply:
I checked with other implementations of S3 and agree with you @MalteJ that it's better to remove the field for now.